vendor/bat-native-ads/src/bat/ads/internal/database/database_migration.cc

```diff
@@ -43,6 +44,10 @@ void Migration::FromVersion(const int from_version, ResultCallback callback) {
     return;
   }
 
+  // TODO(https://github.com/brave/brave-browser/issues/14728): Decouple catalog
+  // business logic once we have implemented database observers
+  ResetCatalog();
```

Comment:
Database migration deletes the catalog database tables, therefore we need to reset the catalog to force a new catalog to be fetched after upgrade